### ISSUE TITLE
VolunteersJobListingFS Errors when the Job has more than 200 related shifts

### ISSUE THREAD

**Issue by jesslopez-sf** [SalesforceFoundation/Volunteers-for-Salesforce#359]
When there are > 200 Volunteer Shifts associated with a Volunteer Job on the VolunteersJobListingFS page, an “invalid query locator” error is returned. If the page is being displayed via a Site, the generic “Authorization Required” error occurs. I believe this is related to a Known Issue currently In Review:
invalid query locator error occurs when iterating child objects in parent-child relationship query result (https://success.salesforce.com/issues_view?id=a1p300000008YGjAAM&title=invalid-query-locator-error-occurs-when-iterating-child-objects-in-parent-child-relationship-query-result)

This does not occur in the regular VolunteersJobListing Visualforce/Apex (only the FS page/controller).

**Debug Log Info**

```
14:02:44:621 FATAL_ERROR System.QueryException: invalid query locator
14:02:44:000 FATAL_ERROR Class.Database.QueryLocatorIterator.hasNext: line 41, column 1
14:02:44:000 FATAL_ERROR Class.GW_Volunteers.VOL_SharedCode.dateTimeFixup: line 128, column 1
14:02:44:000 FATAL_ERROR Class.GW_Volunteers.VOL_CTRL_VolunteersJobListingFS.__sfdc_listVolunteerJobs: line 337, column 1
```
 
**Steps to Repeat:**

1. Create a Campaign (Name = Test Campaign, Active  = true)
2. Create a Volunteer Job associated with that Campaign (Name = “Job With Lots of Shifts”)
3. Insert Volunteer Shifts associated with that Volunteer Job (sample import file attached to Work Item)
4. In your org, go to the following address and make sure your Volunteer Job Id replaces the placeholder:
   ` /apex/VolunteersJobListingFS?showLocationAddress=0&jobId=REPLACE_WITH_YOURS&showLocationInfo=0&showDescription=1&nMonthsToShow=3`
    This should display a list of Volunteer Jobs with no issues (assuming you're under 200 shifts).
5. Now go to the following address and make sure your Volunteer Job Id replaces the placeholder. Increase the MonthsToShow URL parameter to a higher number, like 10:
   ` /apex/VolunteersJobListingFS?showLocationAddress=0&jobId=REPLACE_WITH_YOURS&showLocationInfo=0&showDescription=1&nMonthsToShow=10`
    This results in an invalid query parameter error.

**Comment by jesslopez-sf** [comment]
**lurch: add

**Comment by LurchTheButler** [comment]
Tracking <a href="https://foundation.my.salesforce.com/a2x80000007fO4DAAU" target="_blank">W-028654</a>

**Comment by mrbelvedere** [comment]
Included in beta release 3.111 (Beta 2)

**Comment by mrbelvedere** [comment]
Included in production release 3.111

**Comment by lmeerkatz** [comment]
Reverted in production release 3.112
